command/bridge/withdraw/withdraw_erc20.go

```diff
@@ -0,0 +1,273 @@
+	}
+
+	withdrawCmd.Flags().StringVar(
+		&wp.TxnSenderKey,
+		common.SenderKeyFlag,
+		"",
+		"hex encoded private key of the account which sends withdraw transactions",
+	)
+
+	withdrawCmd.Flags().StringSliceVar(
+		&wp.Receivers,
+		common.ReceiversFlag,
+		nil,
+		"receiving accounts addresses on the root chain",
+	)
+
+	withdrawCmd.Flags().StringSliceVar(
+		&wp.Amounts,
+		common.AmountsFlag,
+		nil,
+		"amounts to send to receiving accounts",
+	)
+
+	withdrawCmd.Flags().StringVar(
+		&wp.childPredicateAddr,
+		childPredicateFlag,
+		contracts.ChildERC20PredicateContract.String(),
+		"ERC20 child chain predicate address",
+	)
+
+	withdrawCmd.Flags().StringVar(
+		&wp.childTokenAddr,
+		childTokenFlag,
+		contracts.NativeERC20TokenContract.String(),
+		"ERC20 child chain token address",
+	)
+
+	withdrawCmd.Flags().StringVar(
+		&wp.jsonRPCAddress,
+		jsonRPCFlag,
+		"http://127.0.0.1:9545",
+		"the JSON RPC child chain endpoint",
+	)
+
+	withdrawCmd.MarkFlagRequired(common.SenderKeyFlag)
+	withdrawCmd.MarkFlagRequired(common.ReceiversFlag)
+	withdrawCmd.MarkFlagRequired(common.AmountsFlag)
+
+	return withdrawCmd
+}
+
+func runPreRunWithdraw(cmd *cobra.Command, _ []string) error {
+	if err := wp.ValidateFlags(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func runCommand(cmd *cobra.Command, _ []string) {
+	outputter := command.InitializeOutputter(cmd)
+	defer outputter.WriteOutput()
+
+	ecdsaRaw, err := hex.DecodeString(wp.TxnSenderKey)
+	if err != nil {
+		outputter.SetError(fmt.Errorf("failed to decode private key: %w", err))
+
+		return
+	}
+
+	key, err := wallet.NewWalletFromPrivKey(ecdsaRaw)
+	if err != nil {
+		outputter.SetError(fmt.Errorf("failed to create wallet from private key: %w", err))
+
+		return
+	}
+
+	txRelayer, err := txrelayer.NewTxRelayer(txrelayer.WithIPAddress(wp.jsonRPCAddress))
+	if err != nil {
+		outputter.SetError(fmt.Errorf("could not create child chain tx relayer: %w", err))
+
+		return
+	}
+
+	exitEventIDs := make([]string, len(wp.Receivers))
+	blockNumbers := make([]string, len(wp.Receivers))
+
+	for i := range wp.Receivers {
+		receiver := wp.Receivers[i]
+		amount := wp.Amounts[i]
+
+		amountBig, err := types.ParseUint256orHex(&amount)
+		if err != nil {
+			outputter.SetError(fmt.Errorf("failed to decode provided amount %s: %w", amount, err))
+
+			return
+		}
+
+		// withdraw tokens transaction
+		txn, err := createWithdrawTxn(types.StringToAddress(receiver), amountBig)
+		if err != nil {
+			outputter.SetError(fmt.Errorf("failed to create tx input: %w", err))
+
+			return
+		}
+
+		receipt, err := txRelayer.SendTransaction(txn, key)
+		if err != nil {
+			outputter.SetError(fmt.Errorf("receiver: %s, amount: %s, error: %w", receiver, amount, err))
+
+			return
+		}
+
+		if receipt.Status == uint64(types.ReceiptFailed) {
+			outputter.SetError(fmt.Errorf("receiver: %s, amount: %s", receiver, amount))
+
+			return
+		}
+
+		exitEvent, blockNumber, err := extractExitEvent(receipt)
```

Comment:
If you are using this `exitEvent` only for it's ID, then maybe you can simplify this `extractExitEvent` function, to only return the ID from the decoded map. WDYT?

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1259/commits/05613644c9cbc998370fba4a89ea67499e7a503a.

Btw introduced `L2StateSynced` event binding, which allowed to simplify decoding exit event logic a bit.